### PR TITLE
Add release note for 2.4.0

### PR DESCRIPTION
**Pull request by heemin32**
Signed-off-by: Heemin Kim <heemin@amazon.com>

### Description
Add release note for 2.4.0
 
### Check List
- [X] New functionality includes testing.
  - [X] All tests pass
- [X] New functionality has been documented.
  - [X] New functionality has javadoc added
- [X] Commits are signed as per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/k-NN/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: release-notes/opensearch-knn.release-notes-2.4.0.0.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
+## Version 2.4.0.0 Release Notes
+
+Compatible with OpenSearch 2.4.0
+
+### Enhancements
+* Merge efficient filtering from feature branch ([#589](https://github.com/opensearch-project/k-NN/pull/589)) [2.x]
```
Comment (naveentatikonda) [inline release-notes/opensearch-knn.release-notes-2.4.0.0.md:6]: I think we will usually list all the PRs that we merged to main branch unless it is merged only to 2.x branch

Comment (heemin32) [reply release-notes/opensearch-knn.release-notes-2.4.0.0.md:6]: Oops. I also need to remove [2.x] part. Forgot that.

**Review by martin-gaievski — COMMENTED** [review]
Just a headsup - we may need to add https://github.com/opensearch-project/k-NN/pull/610 once issues with checks are fixed

**Comment by martin-gaievski** [comment]
> Just a headsup - we may need to add #610 once issues with checks are fixed

#610 is blocked for now by another issue, I've created a manual backport to 2.x https://github.com/opensearch-project/k-NN/pull/613 and 2.4 https://github.com/opensearch-project/k-NN/pull/614

**Comment by heemin32** [comment]
The failure is fixed in 2.x and 2.4 branch. I want to go ahead with force merging.

**Review by martin-gaievski — APPROVED** [review]

**Review by naveentatikonda — APPROVED** [review]

**Pull request merged**